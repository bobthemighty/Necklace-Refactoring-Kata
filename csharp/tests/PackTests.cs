using Microsoft.VisualStudio.TestTools.UnitTesting;
using NecklaceRefactoringKata.Enums;
using NecklaceRefactoringKata.JewelleryTypes;

namespace NecklaceRefactoringKata.Tests
{
    [TestClass()]
    public class PackTests
    {
        [TestMethod()]
        public void Test_EarringStud_IsPackedInTopShelf()
        {
            //arrange
            var jewelleryStorage = new JewelleryStorage();
            var item = new Earring(Type: EarringType.Stud, Stone: Jewel.Amber);
            
            //act
            Packer.Pack(item, jewelleryStorage);

            //assert
            CollectionAssert.Contains(jewelleryStorage.Box.TopShelf, item);
        }

        //To devs: Write more tests here before you refactor
        //note: JewelleryBase types are records, so they automatically have value-based equality checks
        //and can be used with CollectionAssert
    }
}