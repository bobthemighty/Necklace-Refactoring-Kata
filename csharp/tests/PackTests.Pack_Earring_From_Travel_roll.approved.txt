Packing item Earring { Stone = Plain, Type = Drop } (is in travel roll)
Jewellery Storage:
Box:
    Ring Compartment: 
    Top Shelf:        Earring { Stone = Plain, Type = Drop }
    Main Section:     
Tree:                 
Travel Roll:          
Safe:                 
On top of dresser:    
